testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/Environment.java

```diff
@@ -389,6 +419,56 @@ public Builder addContainer(DockerContainer container)
             return this;
         }
 
+        public Builder addConnector(String connectorName)
```

Comment:
nit: `addCatalog` instead of `addConnector`.

BTW if I added a catalog without using this method `selectConfiguredConnectors` should fail right?
But not if I added a password authenticator without this method?

Reply:
I have a little bit of a problem with this, because this method does exactly this - it only registers a connector being used, it does not copy any catalog configs. But the other variants of it do. Can I keep this one and rename others and explain that in a comment?

Regarding password authenticators:
Because currently there's nothing in Trino that would report all configured password authenticators, writing a test that would verify them is pretty hard because it would have to assume some common way of authenticating using every possible way (password, oauth).

Maybe we could add a system table to report this, something like: https://github.com/trinodb/trino/pull/10830

We do pass a list of password authenticators configured in test environments to Tempto, but there's no test to use it: https://github.com/trinodb/trino/pull/10895#discussion_r803904845

Reply:
Let's ignore my comment for the simplicity of overloaded methods.

For password authenticators (and future features) we may end up just checking at review time like we do today when adding a new connector.